src/_P039_Thermosensors.ino

```diff
@@ -276,25 +293,23 @@ boolean Plugin_039(byte function, struct EventStruct *event, String& string)
 
       if (P039_MAX_TYPE == P039_MAX31856) {
 
+
+        // ensure MODE3 access to SPI device
+        SPI.setDataMode(SPI_MODE3);
+
         // init string - content accoring to inital implementation of P039 - MAX31856 read function
-        uint8_t sendBuffer[11] = {0x80, 0x01, (uint8_t) P039_TC_TYPE, 0xFF, 0x7F, 0xC0, 0x7F, 0xFF, 0x80, 0x00, 0x00 };
+        // write to Adress 0x80
+        // activate 50Hz filter in CR0, choose averaging and TC type from configuration in CR1, activate OV/UV/OC faults, write defaults to CJHF, CJLF, LTHFTH, LTHFTL, LTLFTH, LTLFTL, CJTO
+        uint8_t sendBuffer[11] = {0x80, 0x01, (uint8_t) ((P039_CONFIG_4 << 4) | P039_TC_TYPE), 0xFC, 0x7F, 0xC0, 0x7F, 0xFF, 0x80, 0x00, 0x00 };
```

Comment:
Just a warning here...
These C-style cast `(uint8_t)` will work just fine here, but you should try to avoid using them.
Sometimes they can be useful, but they are also kinda tricky as the compiler does not check if it is causing unexpected behavior.
Better use `static_cast<uint8_t>((P039_CONFIG_4 << 4) | P039_TC_TYPE)` here.

Reply:
Understood - this is another one for my programming toolbox. I will search the code for typecasts to migrate for the next push.

Reply:
Solved with next push.